app.py

```diff
@@ -1,36 +1,151 @@
+    if 'COMS3251' not in coursesTaken:
+        needToTake.append(posts.find_one({"course":"COMS3251"})) 
+    if 'COMS3261' not in coursesTaken:
+        needToTake.append(posts.find_one({"course":"COMS3261"})) 
+    if 'CSEE3827' not in coursesTaken:
+        needToTake.append(posts.find_one({"course":"CSEE3827"}))
+    return needToTake
+        
+    
+
+def load_json_to_database(posts):
+    '''This is for loading new courses to be added'''
+    with open(URL) as data_file:
+        data = json.load(data_file)
+
+    # For each course in json
+    for course in data:
+
+        # COMS only
+        title = course["Course"]
+        if title[:4] != "COMS":
+            continue
 
+        # If the course doesnt exist already:
+        if not posts.find_one({"call_number": course["CallNumber"]}):
+            units = course["NumFixedUnits"]
+            if units:
+                units = "{}.{}".format(units[1], units[2])
+
+            to_add = Course(
+                    course["Instructor1Name"].title(),
+
+                    ("Fall {}".format(course["Term"][:4]) if
+                        course["Term"][4] == "2" else
+                        "Winter {}".format(course["Term"][:4])) if
+                        course["Term"] else "",
+
+                    course["Course"][:8] if course["Course"] else "",
+                    course["Course"][9:] if course["Course"] else "",
+                    course["CourseTitle"],
+                    course["CourseSubtitle"],
+                    course["CallNumber"],
+                    units,
+                    ""
+                    )
+
+            # Is there a way to directly insert the instance?
+            posts.insert_one(to_add.__dict__)
+
+
+@app.route('/', methods=['GET'])
 def getInfo():
-    r = requests.get('http://api.culpa.info/courses/department_id/7')
-    jason_data = json.loads(r.text)
+    posts = db.courses
+    users = db_users.login_info
+
+
+    # Clear the database before
+    # posts.remove()
+
+    #load_json_to_database(posts)
+
+    """
     string = ""
-    for i in range(len(jason_data['courses'])):
-        print(string)
-        string += str(jason_data['courses'][i]['number'])+" "
-    data = jason_data['courses'][0]['number']
+    for course in posts.find():
+        x  =  "{} ({})\n".format(course["course_title"],course["course"])
+        if x not in string:
+            string += x
+            string += "<br />"
+
     return string
+
+    """
+
+    return "loaded"
+
+def register_user(name, password):
+    # Return true on success (is unique name)
+    if not db_users.login_info.find_one({"username" : name}):
+        user = {"username" : name,
+                "password" : password}
+        db_users.login_info.insert_one(user)
+        return True
+    return False
+
+
+def is_correct_password(name, password):
+    user = db_users.login_info.find_one({"username" : name})
+    if user["password"] == password:
+        return True
+    return False
```

Comment:
Also, I feel like this is probably vulnerable to a [timing attack](https://en.wikipedia.org/wiki/Timing_attack). I don't know for sure, but I suspect that Python short-circuits string equality so `"aaaaaaaaaaaaaa" == "baaaaaaaaaaaa"` is faster than `"aaaaaaaaaaaaaaaaaa" == "aaaaaaaaaaaaaaab"`.

(I guess the thing to keep in mind is that security is *really* hard and so you should almost always use a 3rd party library written by experts. When you do roll your own stuff, you need to be super paranoid).